pkg/broker/work_engine.go

```diff
@@ -50,10 +54,65 @@ func (engine *WorkEngine) StartNewAsyncJob(
 	if token == "" {
 		token = engine.Token()
 	}
-	go work.Run(token, engine.topic(topic))
+	go engine.startJob(token, work, topic)
+
 	return token, nil
 }
 
+func waitForNotify(sub WorkSubscriber, msg JobMsg, signal chan<- struct{}) {
+	sub.Notify(msg)
+	signal <- struct{}{}
+}
+
+func (engine *WorkEngine) startJob(token string, work Work, topic WorkTopic) {
+	// create a channel specifically for use with this job
+	jobChannel := make(chan JobMsg, engine.jobBufferSize)
+	engine.jobs[token] = jobChannel
+	// ensure we always clean up
+	defer func() {
+		log.Debug("closing channel for job ", token, engine.jobs)
+		close(jobChannel)
+		delete(engine.jobs, token)
+	}()
+
+	go func() {
+		// listen for a new message for the job keyed to this token and hand off to the subscribers async. Wait for them all to be done before accepting
+		// the next message
```

Comment:
reformat this comment to be a couple shorter lines. The first one is really long.